packages/ui/src/contexts/MultiProxyContext.tsx

```diff
@@ -64,136 +76,116 @@ const MultiProxyContextProvider = ({ children }: MultisigContextProps) => {
     [selectedMultiProxy, ownAddressList]
   )
 
-  const refreshAccounList = useCallback((data: MultisigsByAccountsSubscription | null) => {
-    // we do have an answer, but there is no multiproxy
+  const refreshPureToQueryAndMultisigList = useCallback(
+    (data: MultisigsBySignatoriesOrWatchedSubscription | null) => {
+      // we do have an answer, but there is no multisig
+      if (!!data?.accountMultisigs && data.accountMultisigs.length === 0) {
+        setPureToQuery([])
+        setMultisigList([])
+      }
+
+      if (!!data?.accountMultisigs && data.accountMultisigs.length > 0) {
+        const pureToQuerySet = new Set<string>()
+        const multisigMap = new Map<string, MultiProxy>()
+
+        // iterate through the multisigs and populate the multiproxy map
+        // and the list of pure to query
+        data.accountMultisigs.forEach(({ multisig }) => {
+          // if the multisig is a delegatee for at least one pure
+          if (multisig.delegateeFor.length > 0) {
+            // add the pures to the list to query if they are pure proxies
+            multisig.delegateeFor.forEach(({ delegator }) => {
+              delegator.isPureProxy && pureToQuerySet.add(delegator.address)
+            })
+          } else {
+            // if this multisig doesn't have a proxy
+            // we add it to the map and it'll be in the multiproxylist right away
+            multisigMap.set(multisig.address, {
+              proxy: undefined,
+              multisigs: [
+                {
+                  address: multisig.address,
+                  signatories: getSignatoriesFromAccount(multisig.signatories),
+                  threshold: multisig.threshold
+                }
+              ]
+            } as MultiProxy)
+          }
+        })
+
+        // flatten out multisigMap
+        const multisigArray = Array.from(multisigMap.entries()).map(([, multip]) => multip)
```

Comment:
```suggestion
        const multisigArray = Array.from(multisigMap.values())
```
wouldn't that do same thing without allocating array twice? :)

Reply:
nice catch, yes